dashboards-observability/public/components/synthetics/helpers/test_details_flyout.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    EuiButton,
+    EuiComboBox,
+    EuiComboBoxOptionOption,
+    EuiFlexGroup,
+    EuiFlexItem,
+    EuiFlyoutBody,
+    EuiFlyoutFooter,
+    EuiFlyoutHeader,
+    EuiLink,
+    EuiMarkdownFormat,
+    EuiSpacer,
+    EuiText,
+    EuiTitle,
+  } from '@elastic/eui';
+  import { PPL_DOCUMENTATION_URL } from '../../../../common/constants/shared';
+  import _ from 'lodash';
+  import React, { useState } from 'react';
+  import { FlyoutContainers } from '../../common/flyout_containers';
+  import { Group1, Group2, Group3 } from '../../common/helpers/ppl_docs/groups';
+  import { overview } from '../../common/helpers/ppl_docs/overview';
+import { Plt } from '../../visualizations/plotly/plot';
+import { CONTENT_SIZE_FIELD, DOWNLOAD_SPEED_FIELD, DOWNLOAD_TIME_FIELD, REDIRECT_TIMING_FIELD, STARTTRANSFER_TIMING_FIELD, TEST_SUITE_NAME_FIELD, TIMING_FIELDS, URL_FIELD } from '../../../../common/constants/synthetics';
+  
+  type Props = {
+    testSuiteObject: Object | undefined;
+    closeFlyout: () => void;
+  };
+  
+  export const TestDetailsFlyout = ({ testSuiteObject, closeFlyout }: Props) => {
+    
+    // contains all the timing fields being shown
+    const allTimingFields = () => {
+      var retval = []
+      retval = TIMING_FIELDS.slice()
+      retval.push(DOWNLOAD_TIME_FIELD)
+      retval.unshift('Content Download')
+      return retval
+    }
+
+    // contains just the timing fields listed in TIMING_FIELDS
+    const allTimingValues = () => {
+      var retval = [];
+      for (var i = 0; i < TIMING_FIELDS.length; i++) {
```

Comment:
Maybe you could do `retval = TIMING_FIELDS.map(field => testSuiteObject[field])` instead of the for loop

Reply:
I was considering this but since there are no performance enhancements and there are exceptions to this mapping that need to be added in afterwards, I'll hold off on including it here/now.